docs/advanced.rst

```diff
@@ -41,10 +41,56 @@ that need it, like the ``ItemWebPage``.
 You are able to implement your own providers in order to extend or override
 current scrapy-poet behavior.
 
+Currently, scrapy-poet is able to inject instances of the following
+classes as *provider* dependencies:
+
+- :class:`~scrapy.http.Request`
+- :class:`~scrapy.http.Response`
+- :class:`~scrapy.crawler.Crawler`
+- :class:`~scrapy.settings.Settings`
+- :class:`~scrapy.statscollectors.StatsCollector`
+
 .. warning::
 
-    Currently, scrapy-poet is only able to inject ``Response`` and
-    :class:`~.DummyResponse` instances as *provider* dependencies.
+    If your provider needs to use a subclass of Response,
+    for example, TextResponse,
+    make sure to annotate your response argument with the parent class type.
```

Comment:
nitpick: it should be annotated with Response, not with a parent class. E.g. for HtmlResponse parent class is TextResponse.